### ISSUE TITLE
BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding

### ISSUE THREAD

**Issue by ju-ki** [ju-ki/paper-reading#2]
## URL
https://arxiv.org/abs/1810.04805
## date
11/10/2018
## Abstract
* 左右の文脈を共同に条件付けを行うことで、ラベル付されていない文から双方向の表現を事前学習するように設計
* ファインチューニングすることで question answeringなどといった多くのタスクで state-of-the-artを達成

**Comment by ju-ki** [comment]
## Introduction
* 事前学習された言語表現を下流のタスクに適応する方法はfeature_basedとfine_tuningの方法がある
* feature_basedのアプローチとしてELMoなどの追加特徴としての事前学習された表現を含むタスク固有のアーキテクチャーを使用するがある
* fine_tuningのアプローチとしてGPTといった、事前学習されたパラメータを単にファインチューニングして下流タスクで学習させたり、最小限のタスク固有のパラメータを使用する方法がある
* これらのアプローチは特にファインチューニングアプローチにおいて事前学習表現のパワーに制限をかけてしまう。
* 従来のモデルは単方向であり事前学習に使用されるアーキテクチャの選択を制限してしまう。
* これらの制限はsentence_levelタスクやQAタスクにおいて有害->双方向のからの文脈の組み込みが重要になってくる
* finu_tunedアプローチをベースとしたBertを提案してこれらの単方向の制約を緩和する
* 二つの手法MLM(masked language model), NSP(next sentence prediction)を使用する
* MLMは入力された単語を無作為にマスク化し文脈に基づいたマスク化された単語の元のボキャブラリidを予測する
* NSPは二文が渡されて、それが連続しているかの判定
* MLMは左右の文脈を融合させた表現を可能にすることで、深層双方向のtransformerを事前学習可能にしている

**Comment by ju-ki** [comment]
## 関連研究1
### Unsupervised Feature-based Approaches
* 事前学習された単語の埋め込みによって大きく精度を上げている。
* 事前学習された単語の埋め込みを得るためには二通りの方法がある。
* 一つ目は左から右での言語モデル
* 二つ目は左右の文脈から単語が間違っているかどうかを識別する
*

**Comment by ju-ki** [comment]
## 関連研究2
### Unsupervised Fin-tuning Approaches
* これらのアプローチの利点は少ないパラメータで学習ができる
* left-to-right modelやauto-encoderといったモデルがpretrainに使われていた

**Comment by ju-ki** [comment]
## 関連研究3
### Transfer Learning from Supervised Data
* 画像認識の分野ではImageNetなどの教師ありデータを用いた事前学習が自然言語でも有効
*  タスクとしては自然言語推論や機械翻訳

**Comment by ju-ki** [comment]
## Bert
bertには二つのステップがある
* pre-training
* finu-tuning

### アーキテクチャー
* 複数の双方向transfomer-encoderを用いたもの
* L=Layer, H=hidden_size, A=self-attention
* BertBase(L=12, H=768, A=12), total_parameter=110M
* BertLarge(L=24, H=1024, A=16), total_parameter=340M
* OpenAI GPTと同じmodel sizeだが、bertは双方向のself-attentionを使用している

### Input/Output Representations
* 入力表現に関して単一の文やQAなどの文章のペアを明確に表現することができる
* 文ごとの最初のtokenは[CLS]というtokenが使われる
* 文章のペアは単一の文脈として統合される
文章を分ける方法として二つある
* [SEP]というtokenを使用する
* それぞれの文に学習済みの埋め込み表現を加える

**Comment by ju-ki** [comment]
## Pre-training Bert
* bertを事前学習する際に文章を右から左、左から右方向への言語モデルの代わりに、二つの教師なし課題を使う
* Masked LMとNext Sentence Prediction
### Mask LM
* 従来の左右の言語モデルではそれぞれの単語のそれ自身を見てしまう。
* 深層双方向表現を学習するために入力をランダムにマスクにする->masked LM(MLM)
* 15%をマスク化しそれにソフトマック関数に適用し予測。
* fine-tuningの際に出てこないmask tokenがある
* この問題を緩和するために常時mask tokenに置き換えるのではなく以下のような方法で解決した
* 80%でmask tokenに置き換える。10%でランダムなtokenに置き換える10%で全く変更しない

### Next Sentence Prediction(NSP)
* QAや自然言語推論のようなタスクでは二つの文章の関係ベースとなってくる
* 二文が連続するかどうかの判定を行う(labeled as IsNext, and as NotNext)
